rust/worker/src/execution/orchestration/compact.rs

```diff
@@ -286,31 +333,44 @@ impl CompactOrchestrator {
             }
         };
 
-        self.num_write_tasks = partitions.len() as i32;
-        for parition in partitions.iter() {
-            let operator = WriteSegmentsOperator::new();
-            let input = WriteSegmentsInput::new(
+        self.num_write_tasks = partitions.len() as i32 * 3;
+        for partition in partitions.iter() {
+            // We must have an atomic that tracks the current offset ID across partitions to guarantee unique offset IDs for every record.
+            // Every partition has a log materializer that is responsible for updating this atomic.
+            // However, because the log materializer returns borrowed values, we currently run the materializer multiple times for each partition (for each segment writer).
+            // This means that only one log materializer run per partition should access and update the atomic shared across all partitions, and the atomics for the other log materializer runs should not be shared.
+            let shared_curr_max_offset_id = self.curr_max_offset_id.clone();
+            let unshared_curr_max_offset_id1 = Arc::new(AtomicU32::new(
+                self.curr_max_offset_id
+                    .load(std::sync::atomic::Ordering::SeqCst),
```

Comment:
Are we using SeqCst?  If we're not using the atomic for anything other than a count, `Relaxed` will do.